addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from .eu_tax_map import EU_TAX_MAP
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    @api.model
+    def map_all_company_taxes(self):
+        if not self.env['ir.config_parameter'].sudo().get_param('l10n_eu_service.taxes_mapped', False):
+            eu_countries = self.env.ref('base.europe').country_ids
+            companies = self.search([]).filtered(lambda c: c.country_id in eu_countries)
+            companies.map_eu_taxes()
+            self.env['ir.config_parameter'].sudo().set_param('l10n_eu_service.taxes_mapped', True)
+
+    def map_eu_taxes(self):
+        eu_countries = self.env.ref('base.europe').country_ids
+        tax_groups = {}
+        for company in self:
+            invoice_repartition_lines, refund_repartition_lines = company._get_repartition_lines_oss()
+            taxes = self.env['account.tax'].search([('type_tax_use', '=', 'sale'),
+                ('amount_type', '=', 'percent'),
+                ('company_id','=', company.id),
+                ('tax_group_id', 'not ilike', 'OSS')])
+            for country in eu_countries - company.account_tax_fiscal_country_id:
+                mapping = []
+                foreign_taxes = {}
+                fpos = self.env['account.fiscal.position'].search([
+                            ('country_id', '=', country.id),
+                            ('company_id', '=', company.id),
+                            ('auto_apply', '=', True),
+                            ('vat_required', '=', False)], limit=1)
+                if not fpos:
+                    fpos = self.env['account.fiscal.position'].create({
+                        'name': _('OSS B2C %s', country.name),
+                        'country_id': country.id,
+                        'company_id': company.id,
+                        'auto_apply': True,
+                    })
+                for domestic_tax in taxes:
+                    tax_amount = EU_TAX_MAP.get((company.account_tax_fiscal_country_id.code, domestic_tax.amount, country.code), False)
+                    if tax_amount and not fpos.tax_ids.filtered(lambda t_id: t_id.tax_src_id == domestic_tax and t_id.tax_dest_id.amount == tax_amount):
```

Comment:
Rather

if tax_amount and not domestic_tax in fpos.tax_ids

If it's already mapped, we don't care with what; we don't edit the existing setup and consider the user knew what he was doing with that.